src/Microsoft.ML.Data/Transforms/NormalizeColumn.cs

```diff
@@ -400,7 +400,7 @@ public static AffineColumnFunction Create(ModelLoadContext ctx, IHost host, Data
                     if (vectorType.ItemType == NumberDataViewType.Double)
                         return Dbl.ImplVec.Create(ctx, host, vectorType);
                 }
-                throw host.ExceptUserArg(nameof(AffineArgumentsBase.Columns), "Wrong column type. Expected: R4, R8, Vec<R4, n> or Vec<R8, n>. Got: {0}.", typeSrc.ToString());
+                throw host.ExceptUserArg(nameof(AffineArgumentsBase.Columns), "Wrong column type. Expected: Single, Double, Vector<Single, n> or Vector<Double, n>. Got: {0}.", typeSrc.ToString());
```

Comment:
>Vector [](start = 124, length = 6)

In other places we write something like "known-size vector of ...". Might be a bit less confusing than having this mysterious "n" here.

Reply:
It doesn't have to be known size, does this one look ok?

---
In reply to: [267931607](https://github.com/dotnet/machinelearning/pull/3046#discussion_r267931607) [](ancestors = 267931607)

Reply:
Oh, you're right, the code above doesn't require the vector to have a known size :-).

But, isn't this a bug? How can a normalizer be defined on a column of unknown size?


---
In reply to: [267945509](https://github.com/dotnet/machinelearning/pull/3046#discussion_r267945509) [](ancestors = 267945509,267931607)

Reply:
Also, I just noticed that this is the method that loads from `ModelLoadContext`, shouldn't it be `ExceptDecode`?


---
In reply to: [267972968](https://github.com/dotnet/machinelearning/pull/3046#discussion_r267972968) [](ancestors = 267972968,267945509,267931607)

Reply:
> Also, I just noticed that this is the method that loads from `ModelLoadContext`, shouldn't it be `ExceptDecode`?
> 
> In reply to: [267972968](https://github.com/dotnet/machinelearning/pull/3046#discussion_r267972968) [](ancestors = 267972968,267945509,267931607)

Not really. Except decode is for faults of format, which this is not. What this is describing is a properly deserialzed and formatted model being applied to user data, and that user data is not compatible with the transform. So, it would be inappropriate to throw an error explicitly for this.